Sources/Catchable.swift

```diff
@@ -45,8 +45,11 @@ public class PMKFinalizer {
     let pending = Guarantee<Void>.pending()
 
     /// `finally` is the same as `ensure`, but it is not chainable
-    public func finally(_ body: @escaping () -> Void) {
-        pending.guarantee.done(body)
+    public func finally(on: DispatchQueue? = conf.Q.return, flags: DispatchWorkItemFlags? = nil, _ body: @escaping () -> Void) {
+        let fp = pending
+        on.async(flags: flags) {
```

Comment:
Unfortunately this doesn’t work as expected. It immediately dispatches to this queue, we need to pass the queue and flags to the `done`.

Also the test didn't flag this, so we should make it so it does. The reason it doesn't is your check doesn't actually verify which queue we are on, use `dispatchPrecondition(.onQueue(q))` to detect this.

Reply:
> Thanks for the feedback 😃. I tried using dispatchPrecondition(condition: .onQueueAsBarrier(q)) since it was used elsewhere in tests but it wouldn't pass

Well, this proves that it is not executing on the right queue. The assert passes because it is essentially testing if `1 == 1`.